tensorboard/program.py

```diff
@@ -520,21 +521,25 @@ def handle_error(self, request, client_address):
       logger.error('HTTP serving error', exc_info=exc_info)
 
   def get_url(self):
-    if self._auto_wildcard:
-      display_host = socket.gethostname()
-    else:
-      host = self._flags.host
-      display_host = (
+    if not self.display_host:
+
+      if self._auto_wildcard:
+        display_host = socket.getfqdn()
```

Comment:
Just "host" is a better name here since this is still a real hostname; the name "display_host" was meant to be only for the host after adding brackets around an IPv6 address so it renders correctly when printed out in the URL.  See also suggestion above for a `get_url_host()` helper.